password/test/tasty/Validate.hs

```diff
@@ -0,0 +1,236 @@
+{-# LANGUAGE CPP             #-}
+{-# LANGUAGE LambdaCase      #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# OPTIONS_GHC -Wno-orphans #-}
+
+module Validate where
+
+import Control.Monad (replicateM)
+import Data.Char (isAsciiLower, isAsciiUpper, isControl, isDigit, ord)
+import Data.List (nub)
+import Data.Maybe (catMaybes, fromMaybe)
+import Data.Password (mkPassword)
+import Data.Password.Validate (CharSetPredicate (..), CharacterCategory (..),
+                               InvalidReason (..), PasswordPolicy (..),
+                               defaultCharSetPredicate, defaultCharacterSet,
+                               isValidPasswordPolicy, mkCharSetPredicate,
+                               specialCharacterSet, validatePassword)
+import Data.Text (Text)
+import qualified Data.Text as T
+import Test.QuickCheck.Instances.Text ()
+import Test.Tasty (TestTree, testGroup)
+import Test.Tasty.QuickCheck (Arbitrary (..), Gen, Property, choose, elements,
+                              liftArbitrary, oneof, shuffle, suchThat,
+                              testProperty, withMaxSuccess, (===))
+#if! MIN_VERSION_base(4,13,0)
+import Data.Semigroup ((<>))
+#endif
+
+-- | Set of tests used for testing validate module
+testValidate :: TestTree
+testValidate =
+  testGroup
+    "validate"
+    [ testProperty "Length of default character set is valid" $
+        (length $ nub defaultCharacterSet) === 95,
+      testProperty "Length of special character set is valid" $
+        (length $ nub specialCharacterSet) === 33,
+      testProperty "defaultCharacterSet does not contain control characters" $
+        (all (not . isControl) defaultCharacterSet),
+      testProperty "All characters in defaultSet are all ord c < 127" $
+        (all (\c -> ord c < 127) defaultCharacterSet),
```

Comment:
Added test for `defaultCharacterSet` and `specialCharacterSet` which checks the following:

- It does not contain control characters
- Every character applied with `ord` is less than 127
- Cotains 95 unique characters